From: Detlev Casanova <detlev.casanova@collabora.com>
Date: Fri, 20 Sep 2024 10:49:22 -0400
Subject: Revert "fix[drm]: Fix permission check issue on GKI kernel"

On upstream linux, the renderer cannot be used to allocate DUMB buffers,
it has to go through the generic display subsystem, with /dev/dri/card0.

This reverts commit 9c92ab9e20ba33bc9a663ec3b69d3be7c3ed558c.
---
 osal/allocator/allocator_drm.c | 21 ++++-----------------
 osal/mpp_runtime.cpp           |  4 +---
 2 files changed, 5 insertions(+), 20 deletions(-)

diff --git a/osal/allocator/allocator_drm.c b/osal/allocator/allocator_drm.c
index 415dc9b..9bed02f 100644
--- a/osal/allocator/allocator_drm.c
+++ b/osal/allocator/allocator_drm.c
@@ -69,11 +69,7 @@ typedef struct {
     RK_U32  flags;
 } allocator_ctx_drm;
 
-/* use renderD128 first to avoid GKI kernel permission issue */
-static const char *dev_drm[] = {
-    "/dev/dri/renderD128",
-    "/dev/dri/card0",
-};
+static const char *dev_drm = "/dev/dri/card0";
 
 static RK_U32 to_rockchip_gem_mem_flag(RK_U32 flags)
 {
@@ -168,9 +164,8 @@ static int drm_free(int fd, RK_U32 handle)
 
 static MPP_RET os_allocator_drm_open(void **ctx, size_t alignment, MppAllocFlagType flags)
 {
-    allocator_ctx_drm *p;
     RK_S32 fd;
-    RK_S32 i;
+    allocator_ctx_drm *p;
 
     if (NULL == ctx) {
         mpp_err_f("does not accept NULL input\n");
@@ -181,17 +176,9 @@ static MPP_RET os_allocator_drm_open(void **ctx, size_t alignment, MppAllocFlagT
 
     mpp_env_get_u32("drm_debug", &drm_debug, 0);
 
-    for (i = 0; i < (RK_S32)MPP_ARRAY_ELEMS(dev_drm); i++) {
-        fd = open(dev_drm[i], O_RDWR | O_CLOEXEC);
-        if (fd > 0)
-            break;
-    }
-
+    fd = open(dev_drm, O_RDWR | O_CLOEXEC);
     if (fd < 0) {
-        mpp_err_f("open all drm device failed.\n");
-        mpp_err("Please check the following device path and access permission:\n");
-        for (i = 0; i < (RK_S32)MPP_ARRAY_ELEMS(dev_drm); i++)
-            mpp_err("%s\n", dev_drm[i]);
+        mpp_err_f("open %s failed!\n", dev_drm);
         return MPP_ERR_UNKNOW;
     }
 
diff --git a/osal/mpp_runtime.cpp b/osal/mpp_runtime.cpp
index 079463e..c237996 100644
--- a/osal/mpp_runtime.cpp
+++ b/osal/mpp_runtime.cpp
@@ -85,9 +85,7 @@ MppRuntimeService::MppRuntimeService()
 
     allocator_valid[MPP_BUFFER_TYPE_NORMAL] = 1;
     allocator_valid[MPP_BUFFER_TYPE_ION] = !access("/dev/ion", F_OK | R_OK | W_OK);
-    allocator_valid[MPP_BUFFER_TYPE_DRM] =
-        !access("/dev/dri/renderD128", F_OK | R_OK | W_OK) ||
-        !access("/dev/dri/card0", F_OK | R_OK | W_OK);
+    allocator_valid[MPP_BUFFER_TYPE_DRM] = !access("/dev/dri/card0", F_OK | R_OK | W_OK);
     allocator_valid[MPP_BUFFER_TYPE_DMA_HEAP] = !access("/dev/dma_heap", F_OK | R_OK);
 
     if (!allocator_valid[MPP_BUFFER_TYPE_ION] &&
